src/api/app/models/project/update_from_xml_command.rb

```diff
@@ -195,9 +195,22 @@ def update_repository_without_path_element(xml_hash)
     def update_path_elements(current_repo, xml_hash)
       # destroy all current pathelements
       current_repo.path_elements.destroy_all
-      return unless xml_hash['path']
+      return unless xml_hash['path'] || xml_hash['hostsystem']
+
+      # recreate hostsystem elements from xml
+      position = 1
+      xml_hash.elements('hostsystem') do |hostsystem|
+        host_repo = Repository.find_by_project_and_name(hostsystem['project'], hostsystem['repository'])
+        if xml_hash['hostsystem']['project'] == project.name && xml_hash['hostsystem']['repository'] == xml_hash['name']
```

Comment:
careful. You use `hostsystem` as `xml_hash['hostsystem']` and you should continue using that variable to access 'repository' and 'project'
Because  if 'hostsystem' can appear multiple times (as is the topic of your PR), then there is no single xml_hash['hostsystem']['project']

The fact that you didn't spot this yourself, shows a certain lack of test coverage

Reply:
yeah .. that is broken indeed ... (current code is even more broken though as it removes hostsystem even in other repositories)

Reply:
will add a simple test and a fix tomorrow morning, need to finish something else atm